pkg/sparrow/run.go

```diff
@@ -290,13 +307,24 @@ func fanInResults(checkChan chan checks.Result, cResult chan checks.ResultDTO, n
 // shutdown shuts down the sparrow and all managed components gracefully.
 // It returns an error if one is present in the context or if any of the
 // components fail to shut down.
-func (s *Sparrow) shutdown(ctx context.Context) error {
+func (s *Sparrow) shutdown(ctx context.Context) {
 	errC := ctx.Err()
+	log := logger.FromContext(ctx)
 	ctx, cancel := context.WithTimeout(context.Background(), shutdownTimeout)
 	defer cancel()
-	errS := s.targets.Shutdown(ctx)
-	if errS != nil {
-		return fmt.Errorf("failed to shutdown sparrow: %w", errors.Join(errC, errS))
-	}
-	return errC
+
+	s.shutOnce.Do(func() {
+		log.Info("Shutting down sparrow gracefully")
+		var errS error
+		if s.tarMan != nil {
+			errS = s.tarMan.Shutdown(ctx)
+		}
+		errA := s.shutdownAPI(ctx)
+		if errS != nil || errA != nil {
+			log.Error("Failed to shutdown gracefully", "contextError", errC, "apiError", errA, "targetError", errS)
+		}
+
+		// Signal that shutdown is complete
+		s.cDone <- struct{}{}
```

Comment:
Why don't we also shutdown the loader?

Reply:
If the loader has a shutdown, I'll add this tomorrow to the shutdown function. If it doesn't have one, I'll open an issue.

Reply:
https://github.com/caas-team/sparrow/issues/71
I'm still not sure whether we need this. Let's discuss this on Tuesday.